activesupport/lib/active_support/testing/method_call_assertions.rb

```diff
@@ -35,6 +35,53 @@ def assert_not_called(object, method_name, message = nil, &block)
           assert_called(object, method_name, message, times: 0, &block)
         end
 
+        # TODO: No need to resort to #send once support for Ruby 2.4 is
+        # dropped. May even use prepended module. See:
+        # https://github.com/rails/rails/pull/33162#issuecomment-408616175 .
+        def assert_called_on_instance_of(klass, method_name, message = nil, times: 1, returns: nil)
+          times_called = 0
+          klass.send(:define_method, "stubbed_#{method_name}") do |*|
+            times_called += 1
+
+            returns
+          end
+
+          if klass.method_defined?(method_name)
+            klass.send(:alias_method, "original_#{method_name}", method_name)
+          end
+          klass.send(:alias_method, method_name, "stubbed_#{method_name}")
+
+          yield
+
+          error = "Expected #{method_name} to be called #{times} times, but was called #{times_called} times"
+          error = "#{message}.\n#{error}" if message
+
+          assert_equal times, times_called, error
+        ensure
+          if klass.method_defined?("original_#{method_name}")
+            klass.send(:alias_method, method_name, "original_#{method_name}")
+            klass.send(:remove_method, "original_#{method_name}")
+          else
+            klass.send(:remove_method, method_name)
+          end
+          klass.send(:remove_method, "stubbed_#{method_name}")
+        end
+
+        def assert_not_called_on_instance_of(
+          klass,
+          method_name,
+          message = nil,
+          &block
+        )
+          assert_called_on_instance_of(
+            klass,
+            method_name,
+            message,
+            times: 0,
+            &block
+          )
```

Comment:
Let's just keep these arguments on one line, here and elsewhere. Same for the method definition.

Reply:
Fixed.